openff/evaluator/backends/dask_kubernetes.py

```diff
@@ -225,10 +229,34 @@ def __init__(
             assert isinstance(cluster_kwargs, dict)
             self._cluster_kwargs.update(cluster_kwargs)
 
+    def _get_function_key(self, function, args, kwargs) -> str:
+        """Returns a more useful function key
+        
+        Currently all functions passed through Dask are called `wrapped_function`.
+        This returns a key that is either the actual function name, or the
+        Protocol if the function is to execute a protocol.
+        """
+        funckey = funcname(function)
+        if len(args) >= 2:
+            # might be a protocol
+            try:
+                # TODO: might be slow...
+                schema = ProtocolSchema.parse_json(args[1])
+            except Exception:
+                pass
+            else:
+                funckey = f"{schema.type}-{schema.id}"
+        
+        tokenized = tokenize(function, kwargs, *args)
+        return f"{funckey}-{tokenized}"
+
+
     def submit_task(self, function, *args, **kwargs):
         from openff.evaluator.workflow.plugins import registered_workflow_protocols
 
         key = kwargs.pop("key", None)
+        if key is None:
+            key = self._get_function_key(function, args, kwargs)
```

Comment:
I've put this into the DaskKubernetesBackend for now and to test it out, but it would be nice to have upstreamed into the BaseDaskBackend generally.

Reply:
My dashboard is actually informative now instead of a single bar of `_wrapped_function` :-)

<img width="1803" alt="Screenshot 2025-03-28 at 10 40 48 pm" src="https://github.com/user-attachments/assets/378a836c-b8da-4ed3-914a-2763e9f979ec" />